services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenRevocationEndpoint.java

```diff
@@ -242,26 +243,24 @@ private void checkParameterDuplicated(MultivaluedMap<String, String> formParams)
         }
     }
 
-    private void revokeClient() {
-        UserConsentManager.revokeConsentForClient(session, realm, user, client.getId());
+    private void revokeClientSession() {
         if (TokenUtil.TOKEN_TYPE_OFFLINE.equals(token.getType())) {
-            new UserSessionManager(session).revokeOfflineToken(user, client);
+            UserSessionModel userSession = session.sessions().getOfflineUserSession(realm, token.getSessionId());
+            if (userSession != null) {
+                new UserSessionManager(session).removeClientFromOfflineUserSession(realm, userSession, client, user);
+            }
+        }
+        UserSessionModel userSession = session.sessions().getUserSession(realm, token.getSessionId());
```

Comment:
This should be an else, if it's an offline token we don't want to remove the online session. Or am I missing something?

Reply:
Can an offline session and an online session share the same session ID? If not userSession will be null so this will function just like an else. An else would make it more readable though.

Reply:
I think a regular user session is still being created

Reply:
@rmartinc @douglaspalmer @graziang Thanks for the feedback. The online and offline session can share the same ID. I've originally did not updated this part due the previously there was not `else` as well. On the other hand, it will makes sense to add "else" due the:
- Previously the consent was revoked, which effectively revoked both offline and online sessions. But now consent is not revoked and hence makes sense to revoke only one of them
- Revoking only offline session in case that offline-token is provided is more aligned with the recent changes done by @rmartinc regarding offline sessions

So I've added `else` as suggested by @rmartinc to make sure that if offline-token is provided, only related offline-session is revoked, but "online" session is not touched.